snapcraft/plugins/rust.py

```diff
@@ -152,6 +152,13 @@ def _fetch_rust(self):
                   '--disable-sudo', '--save'] + options)
 
     def _fetch_deps(self):
+        
+        source_subdir = getattr(self.options, 'source_subdir', None)
```

Comment:
This could be 

```python
if self.options.source_subdir:
   sourcedir = os.path.join(self.sourcedir, source_subdir)
else:
    sourcedir = self.sourcedir
 ```